esmvalcore/esgf/_search.py

```diff
@@ -97,18 +102,25 @@ def esgf_search_files(facets):
     return files
 
 
-def select_by_time(files, start_year, end_year):
-    """Select files containing data between start_year and end_year."""
+def select_by_time(files, timerange):
+    """Select files containing data between a timerange."""
     selection = []
+    start_date, end_date = _parse_period(timerange)
+
+    start_date = str(start_date)
+    end_date = str(end_date)
```

Comment:
```suggestion
```
It looks like `_parse_period` always returns a tuple of str, so I think this can be deleted?